app/components/EulaModal.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Image, Modal, StyleSheet, TouchableOpacity, View } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+import Markdown from 'react-native-markdown-display';
+
+import closeIcon from './../assets/images/closeIcon.png';
+import { eula_en } from './../locales/eula/eula_en';
+import { eula_es } from './../locales/eula/eula_es';
+import colors from '../constants/colors';
+import Colors from '../constants/colors';
+import ButtonWrapper from './ButtonWrapper';
+import { Checkbox } from './Checkbox';
+import { Typography } from './Typography';
+
+export const EulaModal = props => {
+  const [modalVisible, setModalVisibility] = useState(false);
+  const [boxChecked, toggleCheckbox] = useState(false);
+  const { t } = useTranslation();
+
+  // Pull the EULA in the correct language, with en as fallback
+  let eulaText = eula_en;
+  const eulaLanguages = {
+    en: eula_en,
+    es: eula_es,
+  };
+
+  eulaLanguages[props.selectedLocale] !== undefined
```

Comment:
There is a locale resolver in languages.js that checks device locale against the known locale list. It could be modified to take in a different map and resolve to the best known key in any map with a fallback of English. 

It attempts to handle zh_Hant_HK -> zh-Hant type similar resolutions.

It's partially tested, use that and add some more tests if you think it needs em

Reply:
Will punt on this for now.